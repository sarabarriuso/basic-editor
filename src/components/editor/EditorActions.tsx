import React from 'react';

import t from '../../language';
import Button, { ButtonVariant } from '../common/Button';

interface IEditorActionsProps {
  createShapeCallback: () => void;
  removeShapeCallback: () => void;
  shapeToDelete: boolean;
}

const EditorActions: React.FC<IEditorActionsProps> = ({
  createShapeCallback,
  removeShapeCallback,
  shapeToDelete,
}) => {
  const DEFAULT_CLASSNAME = 'editor-actions';

  return (
    <div className={`${DEFAULT_CLASSNAME}`}>
      <Button
        text={t.text('app.editor.actions.add')}
        onClick={() => {
          createShapeCallback();
        }}
      />
      <Button
        text={t.text('app.editor.actions.remove')}
        onClick={() => {
          removeShapeCallback();
        }}
        disabled={!shapeToDelete}
        variant={ButtonVariant.Secondary}
      />
    </div>
  );
};

export default EditorActions;
