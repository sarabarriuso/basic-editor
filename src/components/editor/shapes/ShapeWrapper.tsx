import React, { useState } from 'react';

// import t from '../../language';
import Shape from './Shape';

const ShapeWrapper: React.FC = () => {
  const DEFAULT_CLASSNAME = 'shape-wrapper';
  const [position, setPosition] = useState({
    x: 0,
    y: 0,
  });
  // const onDragOver = React.useCallback((e) => {
  //   // eslint-disable-next-line no-console
  //   console.log('it dropped!' + e);
  // }, []);

  // const onDragStart = React.useCallback((e) => {
  //   // eslint-disable-next-line no-console
  //   console.log('it started dragging!' + e);
  // }, []);

  // const onDrop = React.useCallback((e) => {
  //   // eslint-disable-next-line no-console
  //   console.log('it was dropped' + e);
  // }, []);

  const handleDragMove = (e: any) => {
    // eslint-disable-next-line no-console
    console.log(position.x + e.movementX);
    setPosition({
      x: position.x + e.movementX,
      y: position.y + e.movementY,
    });
  };

  return (
    <div className={DEFAULT_CLASSNAME}>
      <Shape
        onPointerDown={function (): void {
          // eslint-disable-next-line no-console
          console.log('this si spointer down');
        }}
        onPointerUp={function (): void {
          // eslint-disable-next-line no-console
          console.log('this si spointer up');
        }}
        // I don't think I need this one
        onPointerMove={() => {
          return;
        }}
        onDragMove={handleDragMove}
        styleTransform={{
          transform: `translateX(${position.x}px) translateY(${position.y}px)`,
        }}
      />
    </div>
  );
};

export default ShapeWrapper;
