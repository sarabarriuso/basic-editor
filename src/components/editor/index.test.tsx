import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Editor from '.';
import EditorActions from './EditorActions';
import ShapeWrapper from './shapes/ShapeWrapper';

export default describe('Editor Tests', () => {
  it('Editor renders successfully', () => {
    // Arrange
    // Act
    const tree = shallow(<Editor />);

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(ShapeWrapper).length).toBe(0);
  });

  it('Triggering createShapeCallback creates a new shape', () => {
    // Arrange
    // Act
    const tree = shallow(<Editor />);
    const editorActionsComp = tree.find(EditorActions);
    expect(editorActionsComp).toBeDefined();
    editorActionsComp.prop('createShapeCallback')();

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(ShapeWrapper).length).toBe(1);
  });

  it('Triggering deleteShapeCallback after selecting a shape deletes it', () => {
    // Arrange
    // Act
    const tree = shallow(<Editor />);
    const editorActionsComp = tree.find(EditorActions);

    expect(editorActionsComp).toBeDefined();
    editorActionsComp.prop('createShapeCallback')();
    expect(tree.find(ShapeWrapper).length).toBe(1);
    tree.find(ShapeWrapper).prop('onPointerDown')();
    tree.find(EditorActions).prop('removeShapeCallback')();

    //Assert
    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.find(ShapeWrapper).length).toBe(0);
  });
});
