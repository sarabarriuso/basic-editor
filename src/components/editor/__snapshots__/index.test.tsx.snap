// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Tests Editor renders successfully 1`] = `
<div
  className="editor"
>
  <EditorActions
    createShapeCallback={[Function]}
    removeShapeCallback={[Function]}
    shapeToDelete={false}
  />
  <div
    className="editor__playground"
  />
</div>
`;

exports[`Editor Tests Triggering createShapeCallback creates a new shape 1`] = `
<div
  className="editor"
>
  <EditorActions
    createShapeCallback={[Function]}
    removeShapeCallback={[Function]}
    shapeToDelete={false}
  />
  <div
    className="editor__playground"
  >
    <ShapeWrapper
      isSelected={false}
      key="0"
      onPointerDown={[Function]}
    />
  </div>
</div>
`;

exports[`Editor Tests Triggering deleteShapeCallback after selecting a shape deletes it 1`] = `
<div
  className="editor"
>
  <EditorActions
    createShapeCallback={[Function]}
    removeShapeCallback={[Function]}
    shapeToDelete={false}
  />
  <div
    className="editor__playground"
  />
</div>
`;
